### PR TITLE
Pwm build test

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
based on PR #36181

File: tests/drivers/build_all/pwm/testcase.yaml [hunk 1]
```diff
@@ -1,8 +1,74 @@
...
+    tags: pwm_mcux_ftm
+    extra_args: "CONFIG_PWM_MCUX_FTM=y"
+  drivers.pwm.mcux.pwt.build:
+    platform_allow: twr_ke18f
+    tags: pwm_mcux_pwt
+    extra_args: "CONFIG_PWM_MCUX_PWT=y CONFIG_PWM_CAPTURE=y"
+  drivers.pwm.mcux.ftm.build:
+    platform_allow: frdm_kw41z
+    tags: pwm_mcux_ftm
+    extra_args: "CONFIG_PWM_MCUX_TPM=y"
```
Comment (henrikbrixandersen) [inline tests/drivers/build_all/pwm/testcase.yaml:28]: s/ftm/tpm/g

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: tests/drivers/build_all/pwm/testcase.yaml [hunk 2]
```diff
@@ -1,8 +1,74 @@
...
+  drivers.pwm.sifive.build:
+    platform_allow: hifive1_revb
+    tags: pwm_sifive
+    extra_args: "CONFIG_PWM_SIFIVE=y"
+  drivers.pwm.npcx.build:
+    platform_allow: npcx7m6fb_evb
+    tags: pwm_npcx
+    extra_args: "CONFIG_PWM_NPCX=y"
+  drivers.pwm.nrf.sw.build:
+    platform_allow: nrf52833dk_nrf52820
```
Comment (anangl) [inline tests/drivers/build_all/pwm/testcase.yaml:46]: Please use `nrf51dk_nrf51422` instead, as this driver is mostly used on nRF51, where the hardware PWM is not available.

File: tests/drivers/build_all/pwm/testcase.yaml [hunk 3]
```diff
@@ -1,8 +1,74 @@
...
+    tags: pwm_nrf5_sw
+    extra_args: "CONFIG_PWM_NRF5_SW=y"
+  drivers.pwm.nrf.build:
+    platform_allow: nrf21540dk_nrf52840
```
Comment (anangl) [inline tests/drivers/build_all/pwm/testcase.yaml:50]: `nrf52840dk_nrf52840` will be more appropriate in the PWM context (nRF21540 is an RF front end module).

**Review by anangl — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**